### PR TITLE
Fix getting wrong shape during resizing

### PR DESCRIPTION
**Pull request by sungchul1**
### Summary

ticket no. : 144473
After https://github.com/openvinotoolkit/training_extensions/pull/3582, if `img` is tensor wrong image shape is obtained during resize.
https://github.com/openvinotoolkit/training_extensions/blob/a62e09d4170b8fa48f232e97ff07bc8b7c6a0d4c/src/otx/core/data/transform_libs/torchvision.py#L533

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/core/data/transform_libs/utils.py [hunk 1]
```diff
@@ -104,14 +104,29 @@ def __get__(self, obj, cls):  # noqa: ANN001
...
+    if isinstance(img, Tensor):
+        return img.shape[-2:]
+    if isinstance(img, list):
+        return get_image_shape(img[0])
+
+    msg = f"{type(img)} is not supported."
+    raise TypeError(msg)
+
+
+def to_np_image(img: np.ndarray | Tensor | list) -> np.ndarray:
```
Comment (eunwoosh) [inline src/otx/core/data/transform_libs/utils.py]:
```suggestion
def to_np_image(img: np.ndarray | Tensor | list) -> np.ndarray | list[np.ndarray]:
```

**Review by eunwoosh — COMMENTED** [review]
Thanks for your work :) I left minor comment. Please take a look.

File: src/otx/core/data/transform_libs/utils.py [hunk 2]
```diff
@@ -104,14 +104,29 @@ def __get__(self, obj, cls):  # noqa: ANN001
         return copy.copy(self)
 
 
-def to_np_image(img: Tensor) -> np.ndarray:
+def get_image_shape(img: np.ndarray | Tensor | list) -> tuple[int, int]:
+    """Get image(s) shape with (height, width)."""
+    if isinstance(img, np.ndarray):
+        return img.shape[:2]
```
Comment (eugene123tw) [inline src/otx/core/data/transform_libs/utils.py:114]: Are you encountering the mypy error `Incompatible return value type (got "tuple[int, ...]", expected "tuple[int, int]")`? I suggest explicitly returning it as `return (img.shape[0], img.shape[1])`

File: src/otx/core/data/transform_libs/utils.py [hunk 3]
```diff
@@ -104,14 +104,29 @@ def __get__(self, obj, cls):  # noqa: ANN001
...
+    if isinstance(img, Tensor):
+        return img.shape[-2:]
+    if isinstance(img, list):
+        return get_image_shape(img[0])
+
+    msg = f"{type(img)} is not supported."
+    raise TypeError(msg)
```
Comment (eugene123tw) [inline src/otx/core/data/transform_libs/utils.py]: According to the type hint, the type error is unreachable. Could you add `isinstance(img, (np.ndarray, Tensor, list))` at the start of the function to ensure type checking?

**Review by eugene123tw — COMMENTED** [review]
@sungchul2 I added my comments. Please take a look. Thanks!

**Comment by eugene123tw** [comment]
@sungchul2 Could you also verify if the changes in this PR are covered by unit tests? If not, could you add a unit test to cover them? Thanks.

File: src/otx/core/data/transform_libs/utils.py (hunk 2 above)
Comment (sungchul1) [reply src/otx/core/data/transform_libs/utils.py:114]: I updated this PR with the latest develop and fixed mypy error about `to_np_image`, not `get_image_shape`, at https://github.com/openvinotoolkit/training_extensions/pull/3644/commits/0db74310b56efc377e0bace0a766226a2e5aa331.

File: src/otx/core/data/transform_libs/utils.py (hunk 3 above)
Comment (sungchul1) [reply src/otx/core/data/transform_libs/utils.py]: Added https://github.com/openvinotoolkit/training_extensions/pull/3644/commits/896cfe473a106ab0c3216fe24aaebd41203974ad

**Comment by sungchul1** [comment]
> @sungchul2 Could you also verify if the changes in this PR are covered by unit tests? If not, could you add a unit test to cover them? Thanks.

Thanks for the suggestion! I updated unit tests to cover this issue, https://github.com/openvinotoolkit/training_extensions/pull/3644/commits/35b0fb5a0a5387cf43920bc6e8815382fa7e9a52.

**Review by eunwoosh — DISMISSED** [review]

**Review by eunwoosh — APPROVED** [review]

**Review by eugene123tw — APPROVED** [review]

**Pull request merged**